public/app/features/transformers/partitionByValues/partitionByValues.ts

```diff
@@ -0,0 +1,65 @@
+import { map } from 'rxjs';
+
+import {
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  PluginState,
+  SynchronousDataTransformerInfo,
+  TransformerRegistryItem,
+} from '@grafana/data';
+
+import { PartitionByValuesEditor } from './PartitionByValuesEditor';
+import { partition } from './partition';
+
+interface PartitionByValuesOptions {
+  fields: string[];
+}
+
+export const partitionByValuesTransformer: SynchronousDataTransformerInfo<PartitionByValuesOptions> = {
+  id: DataTransformerID.partitionByValues,
+  name: 'Partition by values',
+  description: `Will split a combined dataset into multiple series discriminated by unique values in one or more chosen fields.`,
+  defaultOptions: {},
+
+  operator: (options) => (source) =>
+    source.pipe(map((data) => partitionByValuesTransformer.transformer(options)(data))),
+
+  transformer: (options: PartitionByValuesOptions) => {
+    return (data: DataFrame[]) => {
+      let keyField = data[0].fields.find((f) => f.name === 'Gender')!
```

Comment:
The options suggests this could be multiple fields, would tat become a new frame for each combination of values?

Reply:
yes